### DISCUSSION TITLE
请问一下，clion和rustRover 编辑器不识别 #[html_sql("...")] 和 impl_select! 绑定方法，有什么办法解决了

### DISCUSSION THREAD

**Discussion by tpc-ht** [rbatis/rbatis#477] [Q&A] [answered]
相关版本
rbatis: 4.3
rust: 1.74.1
![2eacd68c46dbad00dc34d2aefff5dd2](https://github.com/rbatis/rbatis/assets/59469126/99eabe24-4dde-4c5b-973f-6b52c3e8bed7)
![image](https://github.com/rbatis/rbatis/assets/59469126/a91d50ac-baeb-49a4-a1a6-162957547ecd)

目前项目运行没有问题！

**Comment by zhuxiujia** [comment, marked as answer]
> 相关版本 rbatis: 4.3 rust: 1.74.1 ![2eacd68c46dbad00dc34d2aefff5dd2](https://private-user-images.githubusercontent.com/59469126/294854859-99eabe24-4dde-4c5b-973f-6b52c3e8bed7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ3MDU4MTAsIm5iZiI6MTcwNDcwNTUxMCwicGF0aCI6Ii81OTQ2OTEyNi8yOTQ4NTQ4NTktOTllYWJlMjQtNGRkZS00YzViLTk3M2YtNmI1MmMzZThiZWQ3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA4VDA5MTgzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRiNjg5YzIxMGI0OTY3OTQ2ZWFhYzQ3ODdlMGQ2NTI2YzUwZjY4YzY3ZjNhNDRhYjI1ZWJkMDlkMjAzZTViYzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pV4aTXxPJkYLne24VoXAIx_r4zJ8c7KLloWyA7gfhy4) ![image](https://private-user-images.githubusercontent.com/59469126/294854966-a91d50ac-baeb-49a4-a1a6-162957547ecd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ3MDU4MTAsIm5iZiI6MTcwNDcwNTUxMCwicGF0aCI6Ii81OTQ2OTEyNi8yOTQ4NTQ5NjYtYTkxZDUwYWMtYmFlYi00OWE0LWExYTYtMTYyOTU3NTQ3ZWNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA4VDA5MTgzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNmNWE1MjVjNjJhMzc5NmY4ZWY0NDU5NWM1NWM2MDY5OTNiYTVmZDk5MThiOGFkMThkYTk1M2UxZDlhODliZTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Pdaeokwmd76QxW0g1gq2BBMnzVpwdKKPjvxdsEHA0bc)
> 
> 目前项目运行没有问题！

1 编辑器需要cargo build 生成函数后才会智能提示
2 你需要运行rustup update 更新rust的编译器版本到1.75（最新版本Rbatis 需要支持最新的特性 async trait）
3 你可以改成`htmlsql!()` 宏来实现这个函数，这样编辑器可以直接智能提示，不用等编译完成
